src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -2367,5 +2367,152 @@ private struct StructWithBadIgnoreAttribute
             [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
             public Point_2D_Struct MyBadMember { get; set; }
         }
+
+        private interface IUseCustomConverter { }
+
+        [JsonConverter(typeof(MyCustomConverter))]
+        private struct MyValueTypeWithProperties : IUseCustomConverter
+        {
+            public int PrimitiveValue { get; set; }
+            public object RefValue { get; set; }
+        }
+
+        private class MyCustomConverter : JsonConverter<IUseCustomConverter>
+        {
+            public override bool CanConvert(Type typeToConvert)
+            {
+                return typeof(IUseCustomConverter).IsAssignableFrom(typeToConvert);
+            }
+
+            public override IUseCustomConverter Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) =>
+                throw new NotImplementedException();
```

Comment:
Since `IgnoreDefaultValues` is applied to deserialization, can we have one or two assertions on whether this is called?

Reply:
How is `IgnoreDefaultValues` used during deserialization? My understanding is only the older `IgnoreNullValues` affects deserialization.

Reply:
In either case, I added a test for `IgnoreNullValues`.

Reply:
Yes I meant `IgnoreNullValues`, thanks.